interaction-manager/src/lib_components/draggable.js

```diff
@@ -0,0 +1,185 @@
+import { clamp } from '../utils/utility-functions';
+import actions from './actions';
+
+const AXIS = ['x', 'y'];
+/**
+ * Makes an element draggable
+ */
+class Draggable {
+    /**
+     *
+     * @param {Object} options
+     * @param {string} options.element
+     * @param {string} options.restrictTo
+     * @param {string} options.dragClass
+     * @param {'x'|'y'} options.lockAxis
+     * @param {function} options.onDragStart
+     * @param {function} options.onDragMove
+     * @param {function} options.onDragEnd
+     */
+    constructor(options) {
+        const hash = (Math.random() + 1).toString(36).substring(7);
+
+        this.draggableElements = [];
+        this.draggedElement = null;
+        this.enabled = false;
+
+        this.actionName = `drag-around-${hash}`;
+
+        this.onMouseDown = this.onMouseDown.bind(this);
+        this.onMouseMove = this.onMouseMove.bind(this);
+        this.onMouseUp = this.onMouseUp.bind(this);
+
+        this.restrict = {
+            top: 0,
+            left: 0,
+            right: Infinity,
+            bottom: Infinity,
+        };
+
+        this.options = options;
+
+        this.init();
+    }
+
+    /**
+     * Get the index of the dragged item in the draggableElements
+     */
+    get draggedItemIndex() {
+        return [...this.draggableElements].indexOf(this.draggedElement);

```

Comment:
Why do we need to destruct it? Isn't is already an array?

Reply:
Is it because we later assign it to an HTMLCollection which doesn't have indexOf?

Reply:
Yes, at the point where we can use the getter, this.draggableElements is a HTMLCollection